sigmastate/src/main/scala/org/ergoplatform/validation/SigmaValidationSettingsSerializer.scala

```diff
@@ -19,7 +19,9 @@ object SigmaValidationSettingsSerializer extends SigmaSerializer[SigmaValidation
   }
 
   override def parse(r: SigmaByteReader): SigmaValidationSettings = {
-    val nRules = r.getUIntExact
+    val nRules = r.getUInt().toInt
+    // Note, when nRules < 0 as a result of Int overflow, the loop is empty
+    // deserialization will likely fail elsewhere
```

Comment:
Such comment is confusing, "likely fail elsewhere" does not really says anything, please specify how a transaction where a script has negative id encoded will fail. It it will fail, no need to revert getUIntExact

Reply:
Hm, it turns out this serializer is only used by json encoders and not used in Ergo node.
I suggest to leave it as it is (removed confusing "// deserialization will likely fail elsewhere" part)
Added TODO v5.x